lib/failure_detector.mli

```diff
@@ -0,0 +1,37 @@
+(** Messages sent by the protocol *)
+type message =
+  | Ping
+  | Acknowledge
+  | PingRequest of Address.t
+
+(** Here is the definition of the SWIM protocol *)
+type config = {
+  (* This tis the global protocol period, it should be at least three times the round_trip_time *)
+  protocol_period : int;
+  (* The round trip should be around the 99th percentile *)
+  round_trip_time : int;
+  (* NUmber of peers to pick at each round *)
+  peers_to_ping : int;
+}
+
+(** Type holding all the necessary information for the SWIM protocol *)
+type t = {
+  config : config;
+  acknowledges : (int, unit Lwt_condition.t) Base.Hashtbl.t;
+  mutable sequence_number : int;
+}
```

Comment:
Hashtbl is a mutable structure itself, so it doesn't need to be marked mutable 👍 only needs to be mutable if you need to use <- with it

Reply:
Oh, seems obvious since I used `update` function... Thanks